src/jit/lsra.cpp

```diff
@@ -5682,7 +5682,7 @@ void LinearScan::allocateRegisters()
 #ifdef DEBUG
                 // Under stress mode, don't attempt to allocate a reg to
                 // reg optional ref position, unless it's a ParamDef.
-                if (allocateReg && regOptionalNoAlloc() && (currentRefPosition->refType != RefTypeParamDef))
+                if (allocateReg && regOptionalNoAlloc())
```

Comment:
The above was added because, when I made the various pseudo-RefPositions `RegOptional`, I ran into an issue with making the "this" pointer initial assignment be on the stack.